graphwalker-io/src/test/java/org/graphwalker/io/factory/json/JsonContextFactoryTest.java

```diff
@@ -235,4 +235,10 @@ public void PetClinicWithSimpleMachine() throws IOException {
       logger.debug(e.getName());
     }
   }
+  
+  @Test
+  public void acceptDependencyJsonTest() {
```

Comment:
This test only verifies that the factory accepts a file with suffix .json, which makes this test redundant since this test already exists: `public void JsonContextFactoryTest.acceptJsonTest() `

I think that an appropriate test would be to check the dependency attribute. Like:
```java
  @Test
  public void dependencyJsonTest() throws IOException {
    List<Context> contexts = new JsonContextFactory().create(Paths.get("json/DependencyModel.json"));
    Assert.assertNotNull(contexts);
    Assert.assertThat(contexts.size(), is(1));

    Context context = contexts.get(0);

    Assert.assertThat(context.getModel().getVertices().size(), is(2));
    Assert.assertThat(context.getModel().getEdges().size(), is(4));

    Edge.RuntimeEdge e = (Edge.RuntimeEdge)context.getModel().getElementById("e0");
    Assert.assertThat(e.getDependency(), is(100d));

    e = (Edge.RuntimeEdge)context.getModel().getElementById("e1");
    Assert.assertThat(e.getDependency(), is(100d));

    e = (Edge.RuntimeEdge)context.getModel().getElementById("e2");
    Assert.assertThat(e.getDependency(), is(85d));

    e = (Edge.RuntimeEdge)context.getModel().getElementById("e3");
    Assert.assertThat(e.getDependency(), is(15d));
  }
```

But, this won't work out of the box, since GW does not read the dependency attribute in the json file.You need to add this as well:
```
krikar@krikar-ThinkPad-X240:~/dev/mbt/gw3/graphwalker-project$ git diff graphwalker-io/src/main/java/org/graphwalker/io/factory/json/JsonEdge.java                                                                                                      
diff --git a/graphwalker-io/src/main/java/org/graphwalker/io/factory/json/JsonEdge.java b/graphwalker-io/src/main/java/org/graphwalker/io/factory/json/JsonEdge.java                                                                                    
index 8cd9597..33c2e09 100644                                                                                               
--- a/graphwalker-io/src/main/java/org/graphwalker/io/factory/json/JsonEdge.java                                            
+++ b/graphwalker-io/src/main/java/org/graphwalker/io/factory/json/JsonEdge.java                                            
@@ -47,6 +47,7 @@ public class JsonEdge {                                                                                   
   private List<String> actions;                                                                                            
   private List<String> requirements;                                                                                       
   private Map<String, Object> properties;                                                                                  
+  private double dependency;                                                                                               
   private String sourceVertexId;
   private String targetVertexId;
 
@@ -62,6 +63,7 @@ public class JsonEdge {
     Edge edge = new Edge();
     edge.setId(id);
     edge.setName(name);
+    edge.setDependency(dependency);
     edge.setGuard(new Guard(guard));
 
     if (actions != null) {
@@ -86,6 +88,7 @@ public class JsonEdge {
   public void setEdge(Edge.RuntimeEdge edge) {
     id = edge.getId();
     name = edge.getName();
+    dependency = edge.getDependency();
     if (edge.hasGuard()) {
       guard = edge.getGuard().getScript();
     }
@@ -130,6 +133,8 @@ public class JsonEdge {
       edge.setName(name);
     }
 
+    edge.setDependency(dependency);
+
     if (guard != null) {
       edge.setGuard(new Guard(guard));
     }
```